test/Syntax/diagnostics_verify.swift

```diff
@@ -0,0 +1,15 @@
+// RUN: %target-typecheck-verify-swift -verify-syntax-tree
+// FIXME: Too many "unknown syntax" errors.
+
+if true {
+  try
+  // expected-error@-1 2 {{unknown expression syntax exists in the source}}
+  // expected-error@-2 {{unknown statement syntax exists in the source}}
+} // expected-error {{expected expression}} expected-error {{unknown statement syntax exists in the source}}
+
+if false {
+  [.]
+  // expected-error@-1 {{expected identifier after '.' expression}}
+  // expected-error@-2 2 {{unknown expression syntax exists in the source}}
+  // expected-error@-3 {{unknown statement syntax exists in the source}}
+} // expected-error {{unknown statement syntax exists in the source}}
```

Comment:
We are still making a lot of unnecessary `Unknown` nodes for invalid source.

Reply:
Can we design some rules to remove unnecessary nodes as post-processing? e.g. removing empty `CodeBlockItem`.